agent/consul/discoverychain/compile.go

```diff
@@ -478,17 +521,31 @@ RESOLVE_AGAIN:
 		c.resolvers[target.Service] = resolver
 	}
 
+	// Fetch service defaults if set (may be nil)
+	serviceDefaults := c.entries.GetService(resolver.Name)
+
+	// Work out if SNI have been overridden for this service
+	externalSNI := ""
+	if serviceDefaults != nil && serviceDefaults.ExternalSNI != "" {
```

Comment:
Are we not going to allow modifying the SNI of something at a service instance level? I haven't thought through completely whether that would be good or bad (I have a hunch that instance level SNI is probably not a good UX).

Reply:
Not for now because in our own model SNI needs to be varied based on the discovery path taken (e.g. to identify the intended subset) which an individual instance can't know about so it seems like it would get horribly confusing.